### PR TITLE
General enhancements

### PR DESCRIPTION
**Pull request by galvez**
- Rollup for CJS build from ESM (`npm run dist`)
- Eslint config ~with StandardJS~ (`npm run lint`)
- Add `PromisePool` helper for running jobs

Cheers!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,14 +7,27 @@
   "keywords": [
     "mastodon"
   ],
+  "bin": {
+    "monitor": "./dist/index.js"
```
Comment (galvez) [inline package.json]: @claus Probably want a unique name here if distributing over NPM :)

**Comment by galvez** [comment]
😂😂😂😂😂

**Comment by galvez** [comment]
Updating eslint config, hang on :)

**Comment by claus** [comment]
If it helps: https://github.com/MadeInHaus/prettier-config/blob/master/index.json

**Comment by galvez** [comment]
There, happy? Still a grumpy 140-year old :D

**Comment by claus** [comment]
Seems your weird PromisePool shenanigans in resolveRedirects filters more links than it should, have you tested that?

**Comment by galvez** [comment]
Yep, seemed to work, maybe a good chance to add some Jest goodness too. Will revisit it after lunch. Keep this in the icebox for now :)

**Comment by claus** [comment]
It's failing the Claus Wahlers Unit Test™️ though. Indeed might be a good idea to formalize the CWUTs with JESTs 😜

**Comment by galvez** [comment]
Removed `PromisePool` for now.

**Comment by galvez** [comment]
I'll do a separate PR with the Jest setup I commonly use.

**Comment by claus** [comment]
Still swallows a shit ton of links. U sure that an async map fn is a good idea?

**Comment by galvez** [comment]
Haha, nope, that `async`surely won't work :)

I meant to revert to the original `Promise.all()`. I think it's fixed now, finally.

File: utils/extractLinks.js [hunk 2]
```diff
@@ -100,48 +100,26 @@ function cleanLink(link) {
...
+            if (!response || response instanceof Error) {
+                resolve({ ...link, status: 0, hrefCanonical: link.href });
+                return;
+            }
+            const { status, request } = response;
+            if (request.res && request.res.responseUrl) {
+                const { responseUrl: hrefCanonical } = request.res;
+                if (hrefCanonical !== link.href) {
+                    resolve({ ...link, status, hrefCanonical });
+                }
```
Comment (claus) [inline utils/extractLinks.js]:
```suggestion
                    return;
                }
```

File: utils/extractLinks.js [hunk 3]
```diff
@@ -100,48 +100,26 @@ function cleanLink(link) {
...
+            }
```
Comment (claus) [inline utils/extractLinks.js]:
```suggestion
            }
            resolve({ ...link, status: 0, hrefCanonical: link.href });
```

Comment (claus) [reply utils/extractLinks.js]: It was still swallowing links where `hrefCanonical === link.href`.

**Comment by claus** [comment]
Hurry with your stupid lunch i have a PR for your PR 😂

**Comment by galvez** [comment]
Haha, I'm back. Checking the edit suggestions now...

**Comment by galvez** [comment]
K, just applied your edit suggests but haven't retested it yet. I'm gonna open another branch with Jest config and will let the poller run for a while capturing some test input.

**Comment by galvez** [comment]
Added it back in.

**Pull request merged**